homeassistant/components/ecovacs/controller.py

```diff
@@ -81,25 +80,32 @@ async def initialize(self) -> None:
         try:
             devices = await self._api_client.get_devices()
             credentials = await self._authenticator.authenticate()
-            for device_config in devices:
-                if isinstance(device_config, DeviceInfo):
-                    # MQTT device
-                    device = Device(device_config, self._authenticator)
-                    mqtt = await self._get_mqtt_client()
-                    await device.initialize(mqtt)
-                    self._devices.append(device)
-                else:
-                    # Legacy device
-                    bot = VacBot(
-                        credentials.user_id,
-                        EcoVacsAPI.REALM,
-                        self._device_id[0:8],
-                        credentials.token,
-                        device_config,
-                        self._continent,
-                        monitor=True,
-                    )
-                    self._legacy_devices.append(bot)
+            for device_info in devices.mqtt:
+                device = Device(device_info, self._authenticator)
+                mqtt = await self._get_mqtt_client()
+                await device.initialize(mqtt)
+                self._devices.append(device)
+            for device_config in devices.xmpp:
+                bot = VacBot(
+                    credentials.user_id,
+                    EcoVacsAPI.REALM,
+                    self._device_id[0:8],
+                    credentials.token,
+                    device_config,
+                    self._continent,
+                    monitor=True,
+                )
+                self._legacy_devices.append(bot)
+            for device_config in devices.not_supported:
+                _LOGGER.warning(
```

Comment:
I think a repair issue would be helpful here, since it is direct user visible and could link to the troubleshooting section in the docs, were the user can find instructions how to request or contribute the support of the model to the lib?

Reply:
We should only raise repair issues which the user can fix. This is not the case here and therefore I choose to create only a log entry